nix/tools/withTools.nix

```diff
@@ -87,7 +93,7 @@ let
           if test "$_arg_replica" = "on"; then
             replica_slot="replica_$RANDOM"
             replica_dir="$tmpdir/$replica_slot"
-            replica_host="$tmpdir/socket_$replica_slot"
+            replica_host="$tmpdir/sock_$replica_slot"
```

Comment:
We should just run the replica on the same socket path as the primary, but use a different port (it will then create a different socket). That would save a whole level of the path.

Reply:
Great idea! I will try to do these changes in a separate PR.